src/main/java/conversation_history_use_case/MsgSenderRequestModel.java

```diff
@@ -0,0 +1,47 @@
+package conversation_history_use_case;
+
+public class MsgSenderRequestModel {
+    private String senderID;
+    private Object msgContent;
+    private String userID;
+    private String chatID;
+
+    public MsgSenderRequestModel(String senderID, Object msgContent, String userID, String chatID) {
+        this.senderID = senderID;
+        this.msgContent = msgContent;
+        this.userID = userID;
+        this.chatID = chatID;
+    }
+
+    public String getSenderID() {
+        return senderID;
+    }
+
+    public void setSenderID(String senderID) {
+        this.senderID = senderID;
+    }
+
+    public Object getMsgContent() {
+        return msgContent;
+    }
+
+    public void setMsgContent(Object msgContent) {
+        this.msgContent = msgContent;
+    }
+
+    public String getUserID() {
+        return userID;
+    }
+
+    public void setUserID(String userID) {
+        this.userID = userID;
+    }
+
+    public String getChatID() {
+        return chatID;
+    }
+
+    public void setChatID(String chatID) {
+        this.chatID = chatID;
+    }
+}
```

Comment:
Hi James, so from what I can see, it looks like MsgSenderDsRequestModel is the method that will request to add a message on the sender's side and MsgSenderRequestModel is for the recipient's side? Also, what is the difference between ConvHistDsRequestModel and ConvHistRequestModel? Would they also be for sender and recipient side? It's great that you divided up the task into different models though and I like that you added javadoc to make things more clear ~ thank youu

Reply:
Thank you, Amy, for your feedback and questions. I think the name of the classes may not be the clearest -- the classes with "Ds" in the name are data structures that are passed from the interactor to the persistence layer and the classes without "Ds" are passed from the controller to the interactor. The interactor will be called when the sender sends a message; The receiver does not directly use the interactor -- the sender's interactor will update the stored chats' history, and the receiver will see this updated history when they open their app.